compiler/daml-extension/BUILD.bazel

```diff
@@ -3,77 +3,65 @@
 
 package(default_visibility = ["//visibility:public"])
 
-load("@npm_bazel_typescript//:index.bzl", "ts_library")
-load("@build_bazel_rules_nodejs//:defs.bzl", "npm_package")
-load("//bazel_tools:pkg.bzl", "pkg_tar")
-
-ts_library(
-    name = "daml_extension_lib",
-    # TODO(MH): Unfortunately, the current packaging setup does not work with
-    # multiple source files. We need to figure out how to split this file up
-    # before it gets too big and package it properly.
-    srcs = ["src/extension.ts"],
-    node_modules = "@daml_extension_deps//:node_modules",
-    tsconfig = ":tsconfig.json",
-    deps = [
-        "@daml_extension_deps//@types",
-        "@daml_extension_deps//vscode",
-        "@daml_extension_deps//vscode-languageclient",
-    ],
+sh_test(
+    name = "valid-json",
+    srcs = ["ci-tests.sh"],
+    data = glob([
+        "*.json",
+        "syntaxes/*.json",
+    ]),
 )
 
-# With this rule we get access to extension.js, as
-# the ts_library only has the .d.ts file in the outputs.
-# Could possibly also use filegroup, which allows specifying
-# the output group (es5_source).
-npm_package(
-    name = "out",  # named out, so it goes to same place as before
-    deps = [
-        ":daml_extension_lib",
-    ],
+genrule(
+    name = "node_deps_cache",
+    srcs = ["@daml_extension_deps//:node_modules"],
+    outs = ["node_modules.tar.gz"],
+    cmd = """
+        mkdir node_modules
+        cd node_modules
+        for f in $(locations @daml_extension_deps//:node_modules); do
+            file=$$(basename $$f)
+            dir=$$(dirname $$f | sed 's:^.*/node_modules/::')
```

Comment:
Maybe add a comment on what you are doing here. Took me a few seconds to decypher (yes I’m bad at bash :slightly_smiling_face:)

Reply:
Fair enough. It took me a while to figure out I had to do that, too. (Still hoping I'm missing something and I don't actually have to do it, though.)